index.js

```diff
@@ -469,6 +476,28 @@ Router.prototype.use = function use(handler) {
     throw new TypeError('argument handler is required')
   }
 
+  var name
+
+  // If a name is used, the last argument will be a string, not a function
+  if (callbacks.length > 1 && typeof callbacks[callbacks.length - 1] !== 'function') {
+    name = callbacks.pop()
+  }
+
+  if(name && !((path instanceof String) || typeof path === 'string')) {
+    throw new Error('only paths that are strings can be named')
+  }
+
+  if(name && this.routes[name]) {
+    throw new Error('a route or handler with that name already exists')
+  }
+
+  if (name && callbacks.length > 1) {
+    throw new Error('only one handler can be used if Router.use is called with a name parameter')
```

Comment:
This seems like a huge limitation, and I have almost no production app that could ever use the feature like this, because routes need many handlers, like auth handlers, permission handlers, parsing handlers, data loading handlers, etc.